jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/GlobalPipelineMavenConfig.java

```diff
@@ -240,100 +251,109 @@ public synchronized PipelineMavenPluginDao getDao() {
         } else if (j.isTerminating()) {
             throw new IllegalStateException("Request to get DAO whilst Jenkins is terminating");
         }
-        if (dao == null) {
-            try {
-                String jdbcUrl, jdbcUserName, jdbcPassword;
-                if (StringUtils.isBlank(this.jdbcUrl)) {
-                    // default embedded H2 database
-                    File databaseRootDir = new File(j.getRootDir(), "jenkins-jobs");
-                    if (!databaseRootDir.exists()) {
-                        boolean created = databaseRootDir.mkdirs();
-                        if (!created) {
-                            throw new IllegalStateException("Failure to create database root dir " + databaseRootDir);
-                        }
-                    }
-                    jdbcUrl = "jdbc:h2:file:" + new File(databaseRootDir, "jenkins-jobs").getAbsolutePath() + ";" +
-                            "AUTO_SERVER=TRUE;MULTI_THREADED=1;QUERY_CACHE_SIZE=25;JMX=TRUE";
-                    jdbcUserName = "sa";
-                    jdbcPassword = "sa";
-                } else {
-                    jdbcUrl = this.jdbcUrl;
-                    if (this.jdbcCredentialsId == null)
-                        throw new IllegalStateException("No credentials defined for JDBC URL '" + jdbcUrl + "'");
-
-                    UsernamePasswordCredentials jdbcCredentials = (UsernamePasswordCredentials) CredentialsMatchers.firstOrNull(
-                            CredentialsProvider.lookupCredentials(UsernamePasswordCredentials.class, j,
-                                    ACL.SYSTEM, Collections.EMPTY_LIST),
-                            CredentialsMatchers.withId(this.jdbcCredentialsId));
-                    if (jdbcCredentials == null) {
-                        throw new IllegalStateException("Credentials '" + jdbcCredentialsId + "' defined for JDBC URL '" + jdbcUrl + "' NOT found");
+
+        if (dao != null) {
+            return dao;
+        }
+        // if there is no dao and no jdbc url we use NullDao as we do not want to force users
+        // to use jdbc driver with possible security issues
+        if (dao == null && StringUtils.isBlank(jdbcUrl)) {
+            dao = new PipelineMavenPluginNullDao();
+            return dao;
+        }
+
+        try {
+            String jdbcUrl, jdbcUserName, jdbcPassword;
+            if (StringUtils.isBlank(this.jdbcUrl)) {
```

Comment:
Similarly, this clause is now dead code.

I guess you forgot to actually consult `noDataStorage`? It is not used anywhere.